src/Framework/AspNetCoreAnalyzers/test/WebApplicationBuilder/PublicPartialProgramClassTest.cs

```diff
@@ -0,0 +1,195 @@
+    Microsoft.AspNetCore.Analyzers.PublicPartialProgramClassAnalyzer,
+    Microsoft.AspNetCore.Fixers.PublicPartialProgramClassFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers;
+
+public class PublicPartialProgramClassTest
+{
+    [Fact]
+    public async Task DoesNothingWhenNoDeclarationIsFound()
+    {
+        // Arrange
+        var source = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+""";
+
+        // Assert
+        await VerifyCS.VerifyCodeFixAsync(source, source);
+    }
+
+    [Theory]
+    [InlineData("public partial class Program { }")]
+    [InlineData("public partial class Program { /* This is just for tests */ }")]
+    [InlineData("public partial class Program { \n // This is just for tests \n }")]
+    [InlineData("public partial class Program;")]
+    public async Task RemovesDeclarationIfItIsFound(string declarationStyle)
+    {
+        // Arrange
+        var source = $$"""
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+
+{|#0:{{declarationStyle}}|}
+""";
+
+        var diagnostic = new DiagnosticResult(DiagnosticDescriptors.PublicPartialProgramClassNotRequired)
+                .WithLocation(0);
+
+        var fixedSource = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+
+
+""";
+
+        // Assert
+        await VerifyCS.VerifyCodeFixAsync(source, [diagnostic], fixedSource);
+    }
+
+    [Fact]
+    public async Task RemovesDeclarationIfItIsFound_WithLeadingTrivia()
+    {
+        // Arrange
+        var source = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+
+// This is a test
+
+{|#0:public partial class Program;|}
+""";
+
+        var diagnostic = new DiagnosticResult(DiagnosticDescriptors.PublicPartialProgramClassNotRequired)
+                .WithLocation(0);
+
+        var fixedSource = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+
+// This is a test
+
+
+""";
+
+        // Assert
+        await VerifyCS.VerifyCodeFixAsync(source, [diagnostic], fixedSource);
+    }
+
+    [Fact]
+    public async Task DoesNotGeneratesSource_IfProgramDeclaresExplicitInternalAccess()
+    {
+        var source = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+
+internal partial class Program { }
+""";
+
+        await VerifyCS.VerifyCodeFixAsync(source, source);
+    }
+
+    [Fact]
+    public async Task DoesNotFix_ExplicitPublicProgramClass()
+    {
+        var source = """
+using Microsoft.AspNetCore.Builder;
+
+public class Program
```

Comment:
If you make this partial will the analyzer falsely fire?

Reply:
Good catch! Added test coverage and a fix for this.